sdk/core/core/inc/az_result.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_RESULT_H
+#define AZ_RESULT_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum {
+  AZ_OK            =          0,
+  AZ_ERROR_FLAG    = 0x80000000,
+};
+
+enum {
+  AZ_STREAM_FACILITY = 0x1,
+  AZ_JSON_FACILITY   = 0x2,
+};
+
+// The type represents error conditions.
+// Bits:
+// -  0..15 Code.
+// - 16..30 Facility.
+// - 31     Severity (0 - success, 1 - failure).
+typedef int32_t az_result;
+
+#define AZ_MAKE_ERROR(facility, code) ((az_result)(0x80000000 | ((uint32_t)(facility) << 16)) | (uint32_t)(code))
+
+#define AZ_MAKE_RESULT(facility, code) ((az_result)(((uint32_t)(facility) << 16)) | (uint32_t)(code))
+
+inline bool az_failed(az_result result) {
+  return (result & AZ_ERROR_FLAG) != 0;
+}
+
+inline bool az_succedded(az_result result) {
+  return (result & AZ_ERROR_FLAG) == 0;
+}
+
+#define AZ_RETURN_IF_NOT_OK(exp) \
```

Comment:
I don't think we want this. The code should have to do this explicitly if it wants to do this. For example, how can I set a breakpoint on the line where we determined a failure occurred? Also, stepping in a debugger would just jump you around without a clear idea as to why.